### ISSUE TITLE
sync fails in sqlite for table with composite primary key

### ISSUE THREAD

**Issue by fredcy** [sequelize/sequelize#12992]
### What are you doing?

**Here is the link to the SSCCE for this issue:** https://github.com/papb/sequelize-sscce/pull/147 

### What do you expect to happen?

I expected the second sync call to add a `blah` column to the `Demo` table.

### What is actually happening?

In the sqlite v6 case the second `sync()` fails with "UniqueConstraintError [SequelizeUniqueConstraintError]: Validation error".

In my application I found that the equivalent of the `Demo_backup` had each of the primary key columns with UNIQUE status on their own. That does not make sense. The values only need to be over the entire composite key. When the sync tries to copy over rows from Demo to Demo_backup, those row values are not unique in the `foo` column (by test design) and the bogus UNIQUE constraint on `foo` fails hard.

```
  original: [Error: SQLITE_CONSTRAINT: UNIQUE constraint failed: Demo_backup.foo] {
    errno: 19,
    code: 'SQLITE_CONSTRAINT',
    sql: 'INSERT INTO `Demo_backup` SELECT `foo`, `bar` FROM `Demo`;'
  },
```

### Environment

- Sequelize version: sequelize@6.5.0 
- Node.js version: v15.7.0 
- Operating System: MacOS 10.15.7
- If TypeScript related: TypeScript version: na

### How does this problem relate to dialects?

- [ ] I think this problem happens regardless of the dialect.
- [X] I think this problem happens only for the following dialect(s): sqlite 
- [ ] I don't know, I was using PUT-YOUR-DIALECT-HERE, with connector library version XXX and database version XXX

### Would you be willing to resolve this issue by submitting a Pull Request?

- [ ] Yes, I have the time and I know how to start.
- [ ] Yes, I have the time but I don't know how to start, I would need guidance.
- [X] No, I don't have the time, although I believe I could do it if I had the time...
- [ ] No, I don't have the time and I wouldn't even know how to start.

**Comment by GilTos75** [comment]
I'm having exactly the same issue when using sequelize.sync({alter: true}); all "PRIMARY KEY" fields are created as UNIQUE as well.

If I simply use .sync() when the DB doesn't exist it works like a charm, but not useful to sync an existing DB.

**Comment by whyour** [comment]
> This is still an issue

Is this issue solved?

**Comment by ephys** [comment]
I don't think so but it's possible it will be resolved by https://github.com/sequelize/sequelize/pull/14578

**Comment by fedulovivan** [comment]
I can confirm. My case is unique index on two columns defined with:
```js
indexes: [{
  unique: true,
  fields: ['fieldname', 'containerId'],
}]
```
original table contains two coumns without UNIQUE modificator, all as expected:
```sql
-- columns 
`containerId` VARCHAR(255) NOT NULL,
`fieldname` VARCHAR(255) NOT NULL,
-- index
CREATE UNIQUE INDEX `my_table_fieldname_container_id` ON `MyTable` (`fieldname`, `containerId`);
```
after running sync with **alter** option (in my case alteration is adding two new non-related columns)
sequelize creates a new table MyTable_backup and fails to insert into it, since now index has changed:
```sql
-- columns
`containerId` VARCHAR(255) NOT NULL UNIQUE,
`fieldname` VARCHAR(255) NOT NULL UNIQUE,
 -- index
 -- no separate CREATE UNIQUE INDEX directive
```

**Comment by n8vision** [comment]
here's my workaround since this 2-year old bug still isn't fixed even in the latest v7 alpha. this code fully syncs the db per the expected functionality, while retaining keys after the sync:

`const dropAllIndexes = async (db) => {
  const tables = await db.getQueryInterface().showAllTables();

  for(let table of tables){
      const indexes = await db.getQueryInterface().showIndex(table);
      for(let index of indexes){
          await db.getQueryInterface().removeIndex(table, index.name);
      }
  }
};

// Set to true to drop all tables and recreate them. only for dev
const autoMigrate = false;

  if (autoMigrate) {
    await dropAllIndexes(localDB)
  }

  await localDB.sync({ alter: autoMigrate });`

**Comment by TuKun33** [comment]
When you have `Model.sync({ alter: true })` enabled, Sequelize will do the following for each table:
```sql
CREATE TABLE IF NOT EXISTS `YourTable_backup`
INSERT INTO `YourTable_backup` ... FROM `YourTable`;
DROP TABLE `YourTable`
CREATE TABLE IF NOT EXISTS `YourTable`
INSERT INTO `YourTable` ... FROM `YourTable_backup`;
...
```
The problem I encountered was that there was data in the backup table during `sync({ alter: true })`, which caused a primary key conflict when inserting.
So, My solution is to delete the backup table:
```javascript
const sequelize = new Sequelize({...})
const queryInterface = sequelize.getQueryInterface()
const tableNames = await queryInterface.showAllTables();
try {
  const backupTableName = MyModel.tableName + '_backup'
  if (tableNames.includes(backupTableName)) {
    await queryInterface.dropTable(backupTableName)
  }
  await MyModel.sync({ alter: true })
} catch (e) {
  console.error('💔model sync error' , e)
}
```
